code/game/objects/structures/dropship_ammo.dm

```diff
@@ -413,3 +413,132 @@
+
+//Mine laying bomb
+
+/obj/structure/ship_ammo/cas_bomb/minelaying
+	name = "\improper M43 'ML' bomb"
+	desc = "A small, blocky cylinder with a number of smaller tubes strapped to its sides. As the handsome and smart marine you are, you recognize this as a mine laying bomb, meant to dispere its payload over a wide area."
+	icon_state = "default_bomb"
+	ammo_id = "m43"
+	point_cost = 300
+
+/obj/structure/ship_ammo/cas_bomb/minelaying/detonate_on(turf/impact, attackdir = NORTH)
+
+	playsound(impact, 'sound/effects/explosionfar.ogg', 10, 1, 4)
+	impact.visible_message("<span class='notice'>You hear the far away crack of an explosion.</span>")
+
+	impact.ceiling_debris_check(2)
+	addtimer(CALLBACK(src, .proc/drop_mine, impact), 1.5 SECONDS)
+	if(!ammo_count)
+		QDEL_IN(src, travelling_time) ///deleted after the last BOMB has impacted the ground.
+
+/obj/structure/ship_ammo/cas_bomb/minelaying/proc/drop_mine(turf/impact)
+	playsound(loc, 'sound/weapons/guns/interact/mortar_unpack.ogg', 25, 1)
+	var/list/impact_coords = list(list(-3,3),list(0,4),list(3,3),list(-4,0),list(4,0),list(-3,-3),list(0,-4), list(3,-3))
+	var/turf/T
+	var/list/coords
+	var/obj/item/explosive/mine/planted_mine
+	for(var/i=1 to 8)
+		coords = impact_coords[i]
+		T = locate(impact.x+coords[1],impact.y+coords[2],impact.z)
+		T.ceiling_debris_check(2)
+		new/obj/item/explosive/mine(T)
+		planted_mine.attack_self()
+	qdel(src)
+
+///
+///CLUSTER
+///
+
+/obj/structure/ship_ammo/cas_bomb/cluster
+	name = "\improper U402 'Cluster' bomb"
+	desc = "A cluster bomb which blows up before smothering a wide area with bomblets."
+	icon_state = "cluster"
+	ammo_id = "U402"
+	point_cost = 300
+	var/strikes_amount = 18 //the amount of cluster bombs
+
+/obj/structure/ship_ammo/cas_bomb/cluster/detonate_on(turf/impact, attackdir = NORTH)
+	impact.ceiling_debris_check(2)
+	addtimer(CALLBACK(src, .proc/cluster_impacts, impact), 1.5 SECONDS)
+
+/obj/structure/ship_ammo/cas_bomb/cluster/proc/cluster_impacts(turf/impact)
+	for(var/i = 1 to strikes_amount)
+		var/offset_target ///Temp var that is used for the subbomb coordinate
+		offset_target =  get_offset_target_turf(impact, rand(5)-rand(5), rand(5)-rand(5))
+		explosion(offset_target,2,2,1,  adminlog = FALSE)
+
+///
+///White phosphorus
+///
+
+/obj/structure/ship_ammo/cas_bomb/wpsmoke
+	name = "\improper M77 'WP' bomb"
+	desc = "White phosphorus, also known as Willie Pete. Anywhere the bomb hit will be smothered in a blanket of burning smoke."
+	icon_state = "default_bomb"
+	ammo_id = "m77"
+	point_cost = 300
+
+/obj/structure/ship_ammo/cas_bomb/wpsmoke/detonate_on(turf/impact, attackdir = NORTH)
+	impact.ceiling_debris_check(3)
+	explosion(impact,2,2,1, small_animation = TRUE)
+	flame_radius(6, impact, 40, 40) //cooking for a long time
+	var/datum/effect_system/smoke_spread/phosphorus/warcrime = new
+	warcrime.set_up(10, impact, 12)
+	warcrime.start()
+	qdel(src)
+
+//Thermometric bomb
+
+//Welder barrel bomb
+
+//Nail bomb
+
+//toilet-sink
```

Comment:
Ill mark as draft until you finish up

Reply:
@lbnesquik, is welder barrel bomb actually bomb or welder fuel tank (already existing obj i mean)?

Reply:
> 
> 
> @lbnesquik, is welder barrel bomb considered as bomb or welder fuel tank (already existing obj i mean)?

Welder fuel tank with some metal strapped to it. Spread welder fuel before igniting. Or maybe only randomly igniting.